### PR TITLE
Label shift documentation

### PR DESCRIPTION
**Pull request by TammoR**
I have added a tutorial of label shift and empirical risk minimization to the docs. I have also updated the HPO documentation. The HPO default parameter ranges have been updated to search through a few but usually very reasonable settings.

The overall documentation has been restructured by moving SimpleImputer to the beginning and have Encoders and Featurizers that aren't relevant to most users appear later in the text.

--
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source/userguide.rst [hunk 1]
```diff
@@ -245,16 +156,50 @@ The steps for training a model with HPO are identical to the default :code:`Simp
...
+Results for any HPO run can be accessed under :code:`imputer.hpo.results` and the model from any HPO run can then be loaded using :code:`imputer.load_hpo_model(idx)` passing the model index.
+
+
+Label Shift and Empirical Risk Minimization
+*******************************************
+
+The SimpleImputer implements the method described by Lipton, Wang and Smola (https://arxiv.org/abs/1802.03916) to detect and fix label shift for categorical outputs. Label shift occurs when the marginal distribution differs between the training and production setting. For instance, we might be interested in imputing the color of T-Shirts from their free-text description. Let's assume that the training data consists only of women's T-Shirts while the production data consists only of Men's T-Shirts. Then the marginal distribution of colors, p(color), is likely different while the conditional, p(description | color) may be unchanged. This is a scenario where datawig can detect and fix the shift.
```
Comment (tdhd) [inline docs/source/userguide.rst]: Could use a different style of linking references, like:

```
.. _`parameter tutorial`: https://github.com/awslabs/datawig/blob/master/examples/params_tutorial.py
```

File: docs/source/userguide.rst [hunk 2]
```diff
@@ -245,16 +156,50 @@ The steps for training a model with HPO are identical to the default :code:`Simp
...
-    imputer.fit_hpo(
-        train_df=df_train,
-        num_epochs=100,
-        patience=3,
-        learning_rate_candidates=[1e-3, 3e-4, 1e-4]
-    )
+    imputer.fit_hpo(train_df=df_train)
+
+Calling HPO like this will search through some basic and usually helpful hyperparameter choices.
+There are two ways for a more detailed search. Firstly, :code:`fit_hpo` offers additional arguments that can be inspected in the SimpleImputer_. For even more configurations and variation of hyperparameters for the various input column types, a dictionary with ranges can be passed to :code:`fit_hpo` as can be seen in the hpo-code_.
```
Comment (tdhd) [inline docs/source/userguide.rst]: `hpo-code` should point to the public entrypoint of the hyper params dict here: https://github.com/awslabs/datawig/blob/c0d25631e9cda567577d2ed5a34089716831a565/datawig/simple_imputer.py#L167.

File: docs/source/userguide.rst [hunk 3]
```diff
@@ -245,16 +156,50 @@ The steps for training a model with HPO are identical to the default :code:`Simp
...
+Results for any HPO run can be accessed under :code:`imputer.hpo.results` and the model from any HPO run can then be loaded using :code:`imputer.load_hpo_model(idx)` passing the model index.
+
+
+Label Shift and Empirical Risk Minimization
```
Comment (tdhd) [inline docs/source/userguide.rst]: I think having this explanation of label-shift here could confuse readers. Maybe we should place this in a different section where we group advanced features.

Comment (TammoR) [reply docs/source/userguide.rst]: You mean place the entire documentation of this feature in a different section or only the explanation of what label-shift is?
